app/services/report_importer.rb

```diff
@@ -25,19 +32,13 @@ def import
 
     if host.new_record? && !Setting[:create_new_host_when_report_is_uploaded]
```

Comment:
There's duplication between ReportImporter.import and ConfigReporter.import.  Could we either - split this up into smaller, more extensible methods that subclasses can hook into (so we have one code path, jumping to subclass methods to say, parse Puppet-specifics), or remove this version?